ckanext/dia/views.py

```diff
@@ -2,21 +2,56 @@
 from logging import getLogger
 from flask import Blueprint, redirect
 
-log = getLogger(__name__)
+from ckan.logic import ValidationError
+from ckan.plugins import toolkit as tk
+from ckan.common import _, g, request
+from ckan.lib import base
+from ckanext.dia.model import MintedURI
 
+log = getLogger(__name__)
 
-dia = Blueprint("dia", __name__)
 
+no_home_page = Blueprint("no_home_page", __name__)
 
+@no_home_page.route('/', methods=['GET', 'POST'])
 def redirect_to_search():
     return redirect('/dataset')
 
 
-dia.add_url_rule(
-    "/",
-    view_func=redirect_to_search
-)
-
-
-def get_blueprints():
-    return [dia]
+uri_minter = Blueprint("uri_minter", __name__, template_folder='templates')
+
+@uri_minter.before_request
+def before_request():
+    if not g.userobj:
+        base.abort(403, _(u'Not authorized to see this page'))
+
+@uri_minter.route('/uri/new', methods=['GET', 'POST'])
+def new_uri():
+    vars = {'data': {}, 'errors': {}, 'error_summary': {}}
+    if request.method == 'POST':
+        datatype = request.form.get('type')
+        name = request.form.get('name')
+        vars['data'] = request.form
+
+        try:
+            new_instance = MintedURI.create({
+                'type': datatype,
+                'name': name,
+                'created_by_id': g.userobj.id,
+            })
+
+            extra_vars = {
+                'uri': new_instance.uri,
+                'name': new_instance.name,
+            }
+            return tk.render('uris/success.html', extra_vars={'c': extra_vars})
```

Comment:
Is the `c` key needed?

Reply:
I guess it'd more in keeping with the pattern here if I set vars['data'] with the info that needs to display in the template... I think this was modelled off some other templates I looked at.

Reply:
Fixed